glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -670,6 +674,15 @@ class GleanTest {
 
     @Test
     fun `test sending of startup baseline ping with application lifetime metric`() {
+        // Let's tell the MPS that we just sent a metrics ping.
+        // This way we won't trigger it again here.
+        val fakeNowDoNotSend = Calendar.getInstance()
```

Comment:
@badboy why do we need this?

Reply:
From my commit message:

> We don't expect to receive a metrics ping in these tests.
> An "easy" way to prevent it is to mark a metrics ping as recently sent.

For some reason this test (and the other one) suddenly saw a metrics ping before the ping it expected.
I could have also gone and ignore that while waiting for the baseline ping.
I'm not sure _why_ it started to happen now though.

This is the part where I'm unsure about this solution. I just tried to make it work.
The fact that I needed to change other tests could point out that something else is wrong now, or that metrics ping scheduling is very weird there.

Reply:
I _think_ it's worth taking this to a Fenix build and test it out. If it doesn't fix the underlying issue we don't need to waste time to clean up the tests.

If it does, we'll find out what's going on in the tests and find a proper solution.

Reply:
Cool, we can also ask @mcomella for support to double check the fix solves the problem, if that helps. Thanks for tackling this.